test/integration/acceptance/grant_permission_test.cpp

```diff
@@ -0,0 +1,884 @@
+    const std::string
+        source_account_id = source_account_name + "@" + kDomain;
+    const std::string
+        receiver_account_id = receiver_name + "@" + kDomain;
+    const std::string
+        asset_id = IntegrationTestFramework::kAssetName + "#" + kDomain;
+    return TestUnsignedTransactionBuilder()
+        .creatorAccountId(creator_account_id)
+        .createdTime(iroha::time::now())
+        .transferAsset(source_account_id,
+                       receiver_account_id,
+                       asset_id,
+                       "",
+                       amount)
+        .quorum(1)
+        .build()
+        .signAndAddSignature(creator_key)
+        .finish();
+  }
+
+  /**
+   * Get signatories of an account
+   * @param account_name
+   * @param account_key
+   * @return
+   */
+  proto::Query querySignatories(const std::string &account_name,
+                                const crypto::Keypair &account_key) {
+    const std::string
+        account_id = account_name + "@" + kDomain;
+    return proto::QueryBuilder()
+        .creatorAccountId(account_id)
+        .createdTime(iroha::time::now())
+        .queryCounter(1)
+        .getSignatories(account_id)
+        .build()
+        .signAndAddSignature(account_key)
+        .finish();
+  }
+
+  /**
+   * Get account metadata in order to check quorum field
+   * @param account_name
+   * @param account_key
+   * @return
+   */
+  proto::Query queryAccount(const std::string &account_name,
+                            const crypto::Keypair &account_key) {
+    const std::string
+        account_id = account_name + "@" + kDomain;
+    return proto::QueryBuilder()
+        .creatorAccountId(account_id)
+        .createdTime(iroha::time::now())
+        .queryCounter(1)
+        .getAccount(account_id)
+        .build()
+        .signAndAddSignature(account_key)
+        .finish();
+  }
+
+  /**
+   * Get account details
+   * @param account_name
+   * @param account_key
+   * @return
+   */
+  proto::Query queryAccountDetail(const std::string &account_name,
+                                  const crypto::Keypair &account_key) {
+    const std::string
+        account_id = account_name + "@" + kDomain;
+    return proto::QueryBuilder()
+        .creatorAccountId(account_id)
+        .createdTime(iroha::time::now())
+        .queryCounter(1)
+        .getAccountDetail(account_id)
+        .build()
+        .signAndAddSignature(account_key)
+        .finish();
+  }
+
+  /**
+   * Concatenates variadic number of permission vectors
+   * @param perm
+   * @return
+   */
+  auto concatenateVectors(std::initializer_list<std::vector<std::string>> perm) {
+    std::vector<std::string> result;
+    std::vector<std::string> const *p = perm.begin();
+    while (p != perm.end()) {
+      result.insert(result.end(), p->begin(), p->end());
+      p++;
+    }
+    return result;
+  }
+
+  /**
+   * Lambda method that checks query response to contain signature
```

Comment:
Creates a lambda <s>method </s>that checks query response<s>to</s>for <s>contain </s>signatures